src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -491,7 +491,7 @@ void C2_MacroAssembler::string_indexof(Register haystack, Register needle,
   }
   bne(tmp3, skipch, BMSKIP); // if not equal, skipch is bad char
   add(result, haystack, isLL ? nlen_tmp : ch2);
-  ld(ch2, Address(result)); // load 8 bytes from source string
+  load_long_misaligned(ch2, Address(result), ch1); // can use ch1 as tmpreg here as it will be trashed on next mv command anyway
```

Comment:
Suggestion for comment: // can use ch1 as temp register here as it will be trashed by next mv anyway